embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceCacheService.kt

```diff
@@ -292,7 +293,9 @@ internal class EmbraceCacheService(
     }
 
     private fun findLock(name: String) =
-        fileLocks.getOrPut(name, ::ReentrantReadWriteLock)
+        fileLocks[name] ?: synchronized(fileLocks) {
```

Comment:
In practice, I don't know how much contention there would be - an even less lock-y way would be to add some smarts to the default generation function so we don't lock the entire map, but I'm not sure if that would be necessary for this case?